### DISCUSSION TITLE
What is the size limit of a Docusaurus site?

### DISCUSSION THREAD

**Discussion by JenniBeadle** [facebook/docusaurus#9106] [Q&A]
When I tried to add another language to the training manual I have in a Docusaurus site I got an error saying something like at 1,111,000 I had gone **over the 1GB limit**. So I removed the two previous versions and was able to add the extra language.

I now have four active languages (with lots of screenshots) and six more about to start. We want to eventually have 30 languages, but I am concerned that I will reach the limit again well before I have all the languages. (FYI my latest build is 887 MB).

Can someone confirm the actual size limit?

The build size seems excessive for my training manual. It is only 200 A5 pages. I am wondering if the history is taking up a lot of the space. How can I delete the history?

**Comment by slorber** [comment]
There is no "limit" per see, but we recognize Docusaurus is not always the most optimized (https://github.com/facebook/docusaurus/issues/4765).

> I had gone over the 1GB limit. 

Afaik this limit comes with Node.js and you can increase it using settings: that's out of the scope of Docusaurus, and you can just google the error message and see how to increase memory limits of any Node.js software.

One option to do so:

```bash
export NODE_OPTIONS="--max-old-space-size=8192"
```

---

> I now have four active languages (with lots of screenshots) and six more about to start. We want to eventually have 30 languages, but I am concerned that I will reach the limit again well before I have all the languages. (FYI my latest build is 887 MB).

Adding languages is not supposed to increase memory consumption much (unless we have a memory leak). It i possible that one of your language consumes more memory and reaches the Node.js default limit.

You can also replace `yarn build` by building each language independently:

```
yarn build --locale en
yarn build --locale fr
yarn build --locale de
```

If you have many locales it actually makes sense to build each locale in parallel in different CI jobs instead of building them sequentially in a single CI job.

---

> The build size seems excessive for my training manual. It is only 200 A5 pages. I am wondering if the history is taking up a lot of the space. How can I delete the history?

I don't know what you mean by "excessive". What is taking space exactly and what was your expectation? What do you mean by "the history"?
